app/src/main/java/de/tum/in/tumcampusapp/adapters/BarrierfreeFacilityAdapter.java

```diff
@@ -0,0 +1,42 @@
+package de.tum.in.tumcampusapp.adapters;
+
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentStatePagerAdapter;
+
+
+import de.tum.in.tumcampusapp.auxiliary.Const;
+import de.tum.in.tumcampusapp.fragments.BarrierfreeFacilityListFragment;
+
+/**
+ * The adapter used in ViewPager in BarrierFreeFacilitiesActivity.
+ * It passes pageID to BarrierfreeFacilityListFragment to fetch corresponding information.
+ */
+public class BarrierfreeFacilityAdapter extends FragmentStatePagerAdapter{
+
+    private int currentPageID = 0;
+
+    public BarrierfreeFacilityAdapter(FragmentManager fm){
+        super(fm);
+    }
+
+    @Override
+    public Fragment getItem(int position) {
+        Fragment fragment = new BarrierfreeFacilityListFragment();
+        Bundle args = new Bundle();
+        System.out.println(currentPageID);
+        args.putInt(Const.BARRIER_FREE_FACILITY_PAGE_ID, currentPageID);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    public void setCurrentPageID(int currentPageID) {
+        this.currentPageID = currentPageID;
+    }
+
+    @Override
+    public int getCount() {
+        return 1;
```

Comment:
There is one view but the content of the page will be switched according to a spinner.